prep-exercises/b_type_checking.py

```diff
@@ -0,0 +1,51 @@
+# ------------------------
+# Exercise 1
+# ------------------------
+# Read the code to understand what it’s trying to do. 
+# Add type annotations to the method parameters and return types of this code. 
+# Run the code through mypy, and fix all of the bugs that show up. 
+# When you’re confident all of the type annotations are correct, and the bugs are fixed, run the code and check it works.
+
+
+
+# added type annotations to balances of dictionary[string, integer] to represent the structure needed, string to name and integer to amount, 
+# with the return type of none (as there is nothing to return)
+def open_account(balances: dict[str, int], name: str, amount: int) -> None:
+    balances[name] = amount
+
+# added type annotation of dictionary to the accounts argument because it reflects the balances argument,
+# and a return type of integer.
+def sum_balances(accounts: dict[str, int]) -> int:
+    total = 0
+    for name, pence in accounts.items():
+        print(f"{name} had balance {pence}")
```

Comment:
This is a criticism of the problem, not your solution.  It's generally a really bad idea to put print statements inside functions people call to do stuff.  You've no idea where this is being called from so spewing stuff to stdout is... reckless.  Not your issue, problem with the problem, but worth remembering.